packages/evm/test/runCall.spec.ts

```diff
@@ -725,37 +725,67 @@ describe('RunCall tests', () => {
       common,
     })
     for (const opcode of ['f1', 'f2']) {
-      // Code to either CALL or CALLCODE into the own address, with value 1
-      // JUMPDEST added at the start to track how much CALL(CODE)s are made
-      const contractCode = hexToBytes(`0x5B6000808080600130611a90${opcode}`)
-      const contractAddress = Address.fromString('0x000000000000000000000000636F6E7472616374')
-      await evm.stateManager.putContractCode(contractAddress, contractCode)
+      // Code to either CALL or CALLCODE into AACC empty contract, with value 1
 
-      const account = await evm.stateManager.getAccount(contractAddress)
-      account!.balance = BIGINT_100
+      const callCodeAddress = Address.fromString('0x000000000000000000000000000000000000aaaa')
+      const callCode = hexToBytes(`0x6000600060006000600161AACC611a90${opcode}`)
 
-      await evm.stateManager.putAccount(contractAddress, account)
+      const gas = (11600).toString(16).padStart(4, '0') // Enough gas so less than required for CALL/CALLCODE before adding call stipend
+
+      const callerAddress = Address.fromString('0x000000000000000000000000000000000000aaab')
+      const callerCode = hexToBytes(`0x60008080808061AAAA61${gas}f1600055`)
+
+      await evm.stateManager.putAccount(callCodeAddress, new Account())
+      await evm.stateManager.putContractCode(callCodeAddress, callCode)
+
+      await evm.stateManager.putAccount(callerAddress, new Account(undefined, BigInt(1)))
+      await evm.stateManager.putContractCode(callerAddress, callerCode)
 
       const runCallArgs = {
-        gasLimit: BigInt(50000 - 21000),
-        to: contractAddress,
+        to: callerAddress,
+        gasLimit: 0xfffffffn,
       }
+      await evm.runCall(runCallArgs)
+
+      const callResult = bytesToHex(
+        await evm.stateManager.getContractStorage(callerAddress, zeros(32))
+      )
+      // Expect slot to have value of 0 since CALLCODE and CODE should both go OOG
+      assert.equal(callResult, '0x', 'should have result 0 because CALL/CALLCODE should go OOG')
+    }
+
+    for (const opcode of ['f1', 'f2']) {
+      // Code to either CALL or CALLCODE into AACC empty contract, with value 1
 
-      let jumpdestCalls = 0
+      const callCodeAddress = Address.fromString('0x000000000000000000000000000000000000aaaa')
+      const callCode = hexToBytes(`0x6000600060006000600161AACC611a90${opcode}`)
 
-      evm.events.on('step', (e) => {
-        if (e.opcode.name === 'JUMPDEST') {
-          jumpdestCalls++
-        }
-      })
+      const gas = (51600).toString(16).padStart(4, '0') // Enough gas to execute CALL/CALLCODE before adding call stipend
```

Comment:
I think this variant should use the exact amount of gas necessary to correctly perform the CALL(CODE) for explicitness (so when the CALL(CODE) in the callCodeAddress finishes there is exactly 0 gas left, which means it still has enough gas left to finish the call frame)